### PR TITLE
[improve][doc] ConsumerBuilder.java comments

### PR DESCRIPTION
**Pull request by mendonk**
### Motivation
Improve the comments in ConsumerBuilder.java for clarity and cleanliness.
These can be the first documentation devs see in an IDE, so they should inspire more confidence in users. 

### Modifications

Cleaned up the comments in the ConsumerBuilder.java file. 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 1]
```diff
@@ -200,7 +200,7 @@
     ConsumerBuilder<T> ackTimeout(long ackTimeout, TimeUnit timeUnit);
 
     /**
-     * Ack returns receipt but does not mean that the message is not resent after get receipt.
+     * Ack returns receipt, but the message is not re-sent after getting receipt.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:203]: I would use `Acknowledgement`, not `Ack` in docs.

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 2]
```diff
@@ -269,10 +269,10 @@
     ConsumerBuilder<T> subscriptionMode(SubscriptionMode subscriptionMode);
 
     /**
-     * Sets a {@link MessageListener} for the consumer
+     * Sets a {@link MessageListener} for the consumer.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:272]: There's a mix of third-person verbs (eg,. `Sets`) and infinitives (eg. `Select`).
I think it should be made consistent.

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 3]
```diff
@@ -339,22 +339,22 @@
...
-     * message distribution on shared subscription, by pushing messages only to the consumers that are ready to process
+     * <li>Decreases the throughput of the consumer by disabling pre-fetching of messages. This approach improves the
+     * message distribution on shared subscriptions by pushing messages only to the consumers that are ready to process
      * them. Neither {@link Consumer#receive(int, TimeUnit)} nor Partitioned Topics can be used if the consumer queue
      * size is zero. {@link Consumer#receive()} function call should not be interrupted when the consumer queue size is
      * zero.</li>
-     * <li>Doesn't support Batch-Message: if consumer receives any batch-message then it closes consumer connection with
-     * broker and {@link Consumer#receive()} call remains blocked while {@link Consumer#receiveAsync()} receives
-     * exception in callback. <b> consumer is not able to receive any further message unless batch-message in pipeline
+     * <li>Doesn't support Batch-Message. If consumer receives a batch-message, it closes consumer connection with
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:352]:
```suggestion
     * <li>Doesn't support Batch-Message. If a consumer receives a batch-message, it closes the consumer connection with
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 4]
```diff
@@ -339,22 +339,22 @@
...
+     * broker and {@link Consumer#receive()} calls remains blocked while {@link Consumer#receiveAsync()} receives
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:353]:
```suggestion
     * the broker and {@link Consumer#receive()} calls remain blocked while {@link Consumer#receiveAsync()} receives
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 5]
```diff
@@ -339,22 +339,22 @@
...
+     * exception in callback. <b> Consumer is not able to receive any further messages unless batch-message in pipeline
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:354]:
```suggestion
     * exception in callback. <b> The consumer is not able to receive any further messages unless batch-message in pipeline
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 6]
```diff
@@ -339,22 +339,22 @@
...
      * is removed</b></li>
      * </ul>
-     * Default value is {@code 1000} messages and should be good for most use cases.
+     * Default value is {@code 1000} messages and should be adequate for most use cases.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:357]:
```suggestion
     * The default value is {@code 1000} messages and should be adequate for most use cases.
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 7]
```diff
@@ -363,25 +363,25 @@
...
+     *            the max amount of time an acknowledgement can be delayed
      * @param unit
      *            the time unit for the delay
      * @return the consumer builder instance
      */
     ConsumerBuilder<T> acknowledgmentGroupTime(long delay, TimeUnit unit);
 
     /**
-     * Group the consumer acknowledgments for the max size.
+     * Group consumer acknowledgments for the max size.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:382]: Can this be rephrased ?

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 8]
```diff
@@ -477,15 +477,15 @@
...
-     * <p>In Shared subscription mode, broker first dispatches messages to max priority-level
-     * consumers if they have permits, else broker considers next priority level consumers.
+     * <p>In Shared subscription mode, the broker first dispatches messages to max priority-level
+     * consumers if they have permits, otherwise the broker considers next priority level consumers.
      *
-     * <p>If subscription has consumer-A with priorityLevel 0 and Consumer-B with priorityLevel 1
-     * then broker dispatches messages to only consumer-A until it runs out permit and then broker
-     * starts dispatching messages to Consumer-B.
+     * <p>If a subscription has consumer-A with priorityLevel 0 and Consumer-B with priorityLevel 1,
+     * then broker dispatches messages to only consumer-A until it is drained, and then broker will
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:487]:
```suggestion
     * then the broker dispatches messages to only consumer-A until it is drained, and then the broker will
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 9]
```diff
@@ -498,8 +498,8 @@
      * </pre>
      *
      * <p><b>Failover subscription</b>
-     * Broker selects active consumer for a failover-subscription based on consumer's priority-level and
-     * lexicographical sorting of a consumer name.
+     * Broker selects active consumer for a failover subscription based on consumer's priority-level and
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:501]:
```suggestion
     * The broker selects the active consumer for a failover subscription based on consumer's priority-level and
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 10]
```diff
@@ -625,38 +625,39 @@
...
+     * Set the consumer to include the given position of any reset operation like {@link Consumer#seek(long)} or
      * {@link Consumer#seek(MessageId)}}.
      *
      * @return the consumer builder instance
      */
     ConsumerBuilder<T> startMessageIdInclusive();
 
     /**
-     * Set batch receive policy {@link BatchReceivePolicy} for consumer.
+     * Set {@link BatchReceivePolicy} for consumer.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:660]:
```suggestion
     * Set {@link BatchReceivePolicy} for the consumer.
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 11]
```diff
@@ -671,7 +672,7 @@
     ConsumerBuilder<T> batchReceivePolicy(BatchReceivePolicy batchReceivePolicy);
 
     /**
-     * If enabled, the consumer auto retries message.
+     * If enabled, the consumer auto-retries message.
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:675]:
```suggestion
     * If enabled, the consumer auto-retries messages.
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 12]
```diff
@@ -735,8 +736,8 @@
 
     /**
      * Buffering large number of outstanding uncompleted chunked messages can create memory pressure and it can be
-     * guarded by providing this @maxPendingChunkedMessage threshold. Once, consumer reaches this threshold, it drops
-     * the outstanding unchunked-messages by silently acking if autoAckOldestChunkedMessageOnQueueFull is true else it
+     * guarded by providing this @maxPendingChunkedMessage threshold. Once consumer reaches this threshold, it drops
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:739]:
```suggestion
     * guarded by providing this @maxPendingChunkedMessage threshold. Once the consumer reaches this threshold, it drops
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 13]
```diff
@@ -735,8 +736,8 @@
...
+     * the outstanding unchunked-messages by silently acking if autoAckOldestChunkedMessageOnQueueFull is true, else it
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:740]:
```suggestion
     * the outstanding unchunked-messages by silently acknowledging if autoAckOldestChunkedMessageOnQueueFull is true, else it
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 14]
```diff
@@ -747,8 +748,8 @@
     ConsumerBuilder<T> autoAckOldestChunkedMessageOnQueueFull(boolean autoAckOldestChunkedMessageOnQueueFull);
 
     /**
-     * If producer fails to publish all the chunks of a message then consumer can expire incomplete chunks if consumer
-     * won't be able to receive all chunks in expire times (default 1 minute).
+     * If producer fails to publish all the chunks of a message, then consumer can expire incomplete chunks if consumer
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:751]:
```suggestion
     * If the producer fails to publish all the chunks of a message, then the consumer can expire incomplete chunks if the consumer
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 15]
```diff
@@ -747,8 +748,8 @@
...
+     * won't be able to receive all chunks in expire time duration (default 1 minute).
```
Comment (cbornet) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:752]:
```suggestion
     * doesn't receive all chunks during the expire time duration (default 1 minute).
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java (hunk 1 above)
Comment (cbornet) [reply pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:203]: Also I think the should be clarified. As I understand, this is used for transactions. It should be explained what ack receipts are for.

**Review by cbornet — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
```
[INFO] There are 3 errors reported by Checkstyle 8.37 with /home/runner/work/pulsar/pulsar/buildtools/src/main/resources/pulsar/checkstyle.xml ruleset.
Error:  src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:[358] (sizes) LineLength: Line is longer than 120 characters (found 123).
Error:  src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:[744] (sizes) LineLength: Line is longer than 120 characters (found 131).
Error:  src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:[755] (sizes) LineLength: Line is longer than 120 characters (found 131).
```

Please fix CI failure @mendonk.

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 16]
```diff
@@ -140,12 +140,12 @@
     ConsumerBuilder<T> topicsPattern(Pattern topicsPattern);
 
     /**
-     * Specify a pattern for topics that this consumer subscribes on.
+     * Specify a pattern for topics that this consumer subscribes to.
      *
-     * <p>It accepts regular expression that is compiled into a pattern internally. Eg.
+     * <p>Accepts a regular expression that is compiled into a pattern internally. Eg.
```
Comment (momo-jun) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:145]:
```suggestion
     * <p>It accepts a regular expression that is compiled into a pattern internally. E.g.,
```

I don't think changing it to a third-person verb is a good idea. IMO, it makes the annotation even more complex, especially when it's inconsistent throughout the file.

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 17]
```diff
@@ -200,30 +200,34 @@
...
     ConsumerBuilder<T> isAckReceiptEnabled(boolean isAckReceiptEnabled);
 
     /**
      * Define the granularity of the ack-timeout redelivery.
      *
-     * <p>By default, the tick time is set to 1 second. Using an higher tick time
+     * <p>By default, the tick time is set to 1 second. Using a higher tick time
      * reduces the memory overhead to track messages when the ack-timeout is set to
-     * bigger values (eg: 1hour).
+     * bigger values (eg: 1 hour).
```
Comment (momo-jun) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:219]:
```suggestion
     * bigger values (e.g., 1 hour).
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 18]
```diff
@@ -623,40 +627,41 @@
     ConsumerBuilder<T> autoUpdatePartitionsInterval(int interval, TimeUnit unit);
 
     /**
-     * Set KeyShared subscription policy for consumer.
+     * Sets KeyShared subscription policy for consumer.
      *
-     * <p>By default, KeyShared subscription use auto split hash range to maintain consumers. If you want to
-     * set a different KeyShared policy, you can set by following example:
+     * <p>By default, KeyShared subscriptions use auto split hash ranges to maintain consumers. If you want to
+     * set a different KeyShared policy, set a policy by one of the following examples:
```
Comment (momo-jun) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:633]:
```suggestion
     * set a different KeyShared policy, set a policy by using one of the following examples:
```

File: pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java [hunk 19]
```diff
@@ -363,25 +367,25 @@
...
+     *            the max amount of time an acknowledgement can be delayed
      * @param unit
      *            the time unit for the delay
      * @return the consumer builder instance
      */
     ConsumerBuilder<T> acknowledgmentGroupTime(long delay, TimeUnit unit);
 
     /**
-     * Group the consumer acknowledgments for the max size.
+     * Sets number of messages for group consumer acknowledgments.
```
Comment (momo-jun) [inline pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java:386]:
```suggestion
     * Set the number of messages for group consumer acknowledgments.
```
Is "group" necessary and accurate from tech perspective? From the grammar perspective, it should be "grouped" though.

**Review by BewareMyPower — APPROVED** [review]

**Review by momo-jun — APPROVED** [review]

**Comment by tisonkun** [comment]
Merging...

**Pull request merged**